subsys/bluetooth/audio/has.c

```diff
@@ -397,19 +405,20 @@ static struct has_preset {
 	const char *name;
 #endif /* CONFIG_BT_HAS_PRESET_NAME_DYNAMIC */
 	const struct bt_has_preset_ops *ops;
-} has_preset_list[CONFIG_BT_HAS_PRESET_COUNT];
+	sys_snode_t node;
+} preset_pool[CONFIG_BT_HAS_PRESET_COUNT];
 
-/* Number of registered presets */
-static uint8_t has_preset_num;
+static sys_slist_t preset_list = SYS_SLIST_STATIC_INIT(&preset_list);
+static sys_slist_t preset_free_list = SYS_SLIST_STATIC_INIT(&preset_free_list);
```

Comment:
Can you elaborate on why we need two lists?

Reply:
`preset_free_list` is a pool of free preset entries, while `preset_free_list` is the actual list of presets